src/App.tsx

```diff
@@ -1,14 +1,59 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+const defaultTodo = {
+  id: 0,
+  title: 'no title',
+  completed: false,
+  userId: 0,
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoader, setIsLoader] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>(defaultTodo);
+  const [selectValue, setSelectValue] = useState('');
+  const [filterValue, setFilterValue] = useState('');
+
+  useEffect(() => {
+    const getTodosAsync = async () => {
+      const receivedTodos = await getTodos();
+
+      setTodos(receivedTodos);
+      setIsLoader(false);
+    };
+
+    getTodosAsync();
+  }, []);
+
+  const selectTodo = (todo: Todo) => {
+    setSelectedTodo(todo);
+  };
+
+  const hundleSelect = (value: string) => {
+    setSelectValue(value);
+  };
+
+  const hundleChange = (value: string) => {
+    setFilterValue(value);
+  };
+
+  const todosForRender
+    = todos
+      .filter(({ completed }) => (
+        selectValue ? String(completed) === selectValue : String(completed) !== selectValue
+      ))
```

Comment:
Good solution 👍 
But better create a separate variable and use `switch case` to make code more readable.

Reply:
no creativity whatsoever 😢

Reply:
Creativety it is great 👍  

But your code still should be clear and easy to read for other programmers.
In the real project it is great when you use the ternary operator in some easy cases, but for some more complicated things should write easy to read code.